### PR TITLE
Feature: Added support for installing multiple fonts at the same time

### LINKED ISSUE #9295: Feature: Support for installing multiple fonts at the same time
### Description

Whenever I try to install a single font or multiple fonts, the  **Install** option in right-click context menu doesn't work.
image

### Steps To Reproduce

1. Select multiple font type files such as TTF & OTF
2. Right-click -> More -> Install
3. Only the first item in the selection will be installed in the system.

**"Install for all users"** doesn't work too

### Expected behavior

I should be able to install multiple fonts and this dialog should appear
image

### Files Version

2.3.0.0 
Still broken in v2.3.61

### Windows Version

10.0.22621.1

### Relevant Assets/Logs

_N/A_

### PR DESCRIPTION
**Pull request by hecksmosis**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #9295

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

File: src/Files.App/Helpers/ShellContextMenuHelper.cs [hunk 1]
```diff
@@ -173,15 +173,19 @@ async void InvokeShellMenuItem(ContextMenu contextMenu, object? tag)
 					case "install" when isFont:
 						{
 							var userFontDir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "Microsoft", "Windows", "Fonts");
-							var destName = Path.Combine(userFontDir, Path.GetFileName(contextMenu.ItemsPath[0]));
-							Win32API.RunPowershellCommand($"-command \"Copy-Item '{contextMenu.ItemsPath[0]}' '{userFontDir}'; New-ItemProperty -Name '{Path.GetFileNameWithoutExtension(contextMenu.ItemsPath[0])}' -Path 'HKCU:\\Software\\Microsoft\\Windows NT\\CurrentVersion\\Fonts' -PropertyType string -Value '{destName}'\"", false);
+							foreach (string path in contextMenu.ItemsPath)
+							{
+								var destName = Path.Combine(userFontDir, Path.GetFileName(path));
+								Win32API.RunPowershellCommand($"-command \"Copy-Item '{path}' '{userFontDir}'; New-ItemProperty -Name '{Path.GetFileNameWithoutExtension(path)}' -Path 'HKCU:\\Software\\Microsoft\\Windows NT\\CurrentVersion\\Fonts' -PropertyType string -Value '{destName}'\"", false);
```
Comment (yaira2) [inline src/Files.App/Helpers/ShellContextMenuHelper.cs:179]: Can this be merged into a common method? You can pass a parameter to set if it applies to all users.

**Review by QuaintMako — APPROVED** [review]
Minor suggestion, LGTM and works on my machine.

File: src/Files.App/Helpers/ShellContextMenuHelper.cs [hunk 2]
```diff
@@ -199,7 +198,17 @@ async void InvokeShellMenuItem(ContextMenu contextMenu, object? tag)
 						await contextMenu.InvokeItem(menuId);
 						break;
 				}
-
+				
+				void InstallFont(string path, bool asAdmin)
+				{
+					string dir;
```
Comment (yaira2) [inline src/Files.App/Helpers/ShellContextMenuHelper.cs:204]:
```suggestion
					string dir = asAdmin ? Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Windows), "Fonts") : Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "Microsoft", "Windows", "Fonts");
```

File: src/Files.App/Helpers/ShellContextMenuHelper.cs [hunk 3]
```diff
@@ -199,7 +198,17 @@ async void InvokeShellMenuItem(ContextMenu contextMenu, object? tag)
...
 				}
-
+				
+				void InstallFont(string path, bool asAdmin)
+				{
+					string dir = asAdmin ? Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Windows), "Fonts") : Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "Microsoft", "Windows", "Fonts");
+					if (asAdmin)
+						dir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Windows), "Fonts");
+					else
+						dir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "Microsoft", "Windows", "Fonts");
```
Comment (yaira2) [inline src/Files.App/Helpers/ShellContextMenuHelper.cs:208]:
```suggestion
```

**Comment by yaira2** [comment]
@hecksmosis if you're okay with my changes, LGTM.

**Comment by hecksmosis** [comment]
Ok

**Comment by hecksmosis** [comment]
(I actually was making them on my machine, but you beat me to it

**Comment by hecksmosis** [comment]
I think this is done then

File: src/Files.App/Helpers/ShellContextMenuHelper.cs [hunk 4]
```diff
@@ -199,7 +198,14 @@ async void InvokeShellMenuItem(ContextMenu contextMenu, object? tag)
 						await contextMenu.InvokeItem(menuId);
 						break;
 				}
+				
+				void InstallFont(string path, bool asAdmin)
+				{
+					string dir = asAdmin ? Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Windows), "Fonts")
+						: Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "Microsoft", "Windows", "Fonts");
 
+					Win32API.RunPowershellCommand($"-command \"Copy-Item '{path}' '{dir}'; New-ItemProperty -Name '{Path.GetFileNameWithoutExtension(path)}' -Path 'HKCU:\\Software\\Microsoft\\Windows NT\\CurrentVersion\\Fonts' -PropertyType string -Value '{dir}'\"", asAdmin);
```
Comment (gave92) [inline src/Files.App/Helpers/ShellContextMenuHelper.cs:207]: When installing as admin you should use HKLM not HKCU.

Comment (yaira2) [reply src/Files.App/Helpers/ShellContextMenuHelper.cs:207]: Seems we'll need another var for registry path.

Comment (hecksmosis) [reply src/Files.App/Helpers/ShellContextMenuHelper.cs]: Done

**Pull request merged**